editoast/src/views/rolling_stock/light.rs

```diff
@@ -108,6 +114,7 @@ async fn list(
     State(db_pool): State<DbConnectionPoolV2>,
     Extension(authorizer): AuthorizerExt,
     Query(page_settings): Query<PaginationQueryParam>,
+    Query(query): Query<LightRollingStockQuery>,
 ) -> Result<Json<LightRollingStockWithLiveriesCountList>> {
     let authorized = authorizer
         .check_roles([BuiltinRole::RollingStockCollectionRead].into())
```

Comment:
As discussed, we might want to check `BuiltinRole::FreightRollingStockCollectionRead` when `query.freight_compatible == true`, and else, check `BuiltinRole::RollingStockCollectionRead`.